webknossos-datastore/app/com/scalableminds/webknossos/datastore/datavault/DataVault.scala

```diff
@@ -5,6 +5,9 @@ import com.scalableminds.util.tools.Fox
 import scala.concurrent.ExecutionContext
 
 trait DataVault {
+  val MAX_EXPLORED_ITEMS_PER_LEVEL = 10
   def readBytesAndEncoding(path: VaultPath, range: RangeSpecifier)(
       implicit ec: ExecutionContext): Fox[(Array[Byte], Encoding.Value)]
+
+  def listDirectory(path: VaultPath)(implicit ec: ExecutionContext): Fox[List[VaultPath]]
```

Comment:
I’d add a parameter `limit: Option[Int]` here, rather than use the constant, see above.